server/src/main/java/org/apache/iotdb/db/query/reader/chunk/MemAlignedPageReader.java

```diff
@@ -75,44 +70,59 @@ public BatchData getAllSatisfiedPageData(boolean ascending) throws IOException {
       // accept AlignedPath with only one sub sensor
       if (firstNotNullObject != null
           && (valueFilter == null
-              || valueFilter.satisfy(timeValuePair.getTimestamp(), firstNotNullObject))) {
-        batchData.putVector(timeValuePair.getTimestamp(), values);
+              || valueFilter.satisfy(tsBlock.getTimeByIndex(row), firstNotNullObject))) {
+        TsPrimitiveType[] values = new TsPrimitiveType[tsBlock.getValueColumnCount()];
+        for (int column = 0; column < tsBlock.getValueColumnCount(); column++) {
+          if (tsBlock.getColumn(column) != null && !tsBlock.getColumn(column).isNull(row)) {
+            values[column] = tsBlock.getColumn(column).getTsPrimitiveType(row);
+          }
+        }
+        batchData.putVector(tsBlock.getTimeByIndex(row), values);
       }
     }
     return batchData.flip();
   }
 
   @Override
-  public TsBlock getAllSatisfiedData() throws IOException {
-    // TODO change from the row-based style to column-based style
+  public TsBlock getAllSatisfiedData() {
     TsBlockBuilder builder =
         new TsBlockBuilder(
             chunkMetadata.getValueChunkMetadataList().stream()
                 .map(IChunkMetadata::getDataType)
                 .collect(Collectors.toList()));
-    while (timeValuePairIterator.hasNextTimeValuePair()) {
-      TimeValuePair timeValuePair = timeValuePairIterator.nextTimeValuePair();
-      TsPrimitiveType[] values = timeValuePair.getValue().getVector();
-      // save the first not null value of each row
-      Object firstNotNullObject = null;
-      for (TsPrimitiveType value : values) {
-        if (value != null) {
-          firstNotNullObject = value.getValue();
-          break;
-        }
-      }
+
+    boolean[] valueSatisfyInfo = new boolean[tsBlock.getPositionCount()];
+    for (int column = 0; column < tsBlock.getValueColumnCount(); column++) {
+
       // if all the sub sensors' value are null in current time
       // or current row is not satisfied with the filter, just discard it
-      // TODO fix value filter firstNotNullObject, currently, if it's a value filter, it will only
+      // currently, if it's a value filter, it will only
       // accept AlignedPath with only one sub sensor
-      if (firstNotNullObject != null
-          && (valueFilter == null
-              || valueFilter.satisfy(timeValuePair.getTimestamp(), firstNotNullObject))) {
-        builder.getTimeColumnBuilder().writeLong(timeValuePair.getTimestamp());
-        for (int i = 0; i < values.length; i++) {
-          builder.getColumnBuilder(i).writeTsPrimitiveType(values[i]);
+      for (int row = 0; row < tsBlock.getPositionCount(); row++) {
+        if (column == 0) {
+          Object value = tsBlock.getColumn(column).getObject(row);
+          if (value != null && (valueFilter == null || valueFilter.satisfy(0, value))) {
+            builder.getColumnBuilder(column).writeObject(tsBlock.getColumn(column).getObject(row));
```

Comment:
use `ColumnBuilder write(Column column, int index);` in `ColumnBuilder` to avoid boxing and unboxing.